nmtwizard/preprocess/tu.py

```diff
@@ -226,9 +226,8 @@ def append(self, other):
     def replace_tokens(self, start_idx, tok_num, new_tokens=None, part=0):
 
         # check/initialize tokenization if not done already
-        cur_tokens = self.__tok
+        cur_tokens = self.__tok[part]
```

Comment:
I think the part indexing should happen after the `cur_tokens is not None` check.